cpp/log/log_verifier.h

```diff
@@ -33,7 +33,7 @@ class LogVerifier {
     INVALID_MERKLE_PATH,
```

Comment:
When we unify with Status we should maybe have a detail message slot in Status so we can pass the failure reason back to users.